firestore/google/cloud/firestore_v1beta1/_helpers.py

```diff
@@ -119,6 +120,56 @@ def __ne__(self, other):
             return not equality_val
 
 
+class FieldPath(object):
+    """ Field Path object for client use.
+
+    Args:
+        parts (string or iterable): Indicating path of the key to be used.
+    """
+
+    def __init__(self, *parts):
+        invalid_characters = '~*/[]'
+        if len(parts) == 1:
+            parts = parts[0]
+        if isinstance(parts, basestring):
```

Comment:
If this is intended to be part of the public API, then I would like to suggest to separate the two cases here more clearly:

If we pass a string, we split on dots and reject ~*/[]. If we pass a list of strings, we don't.

So, calling FieldPath with (["foo.bar"]) should result in a field path of `foo.bar`. 

The FieldPath constructor should ideally never split on dots.